src/ui/public/Binder/Binder.js

```diff
@@ -0,0 +1,49 @@
+var $ = require('jquery');
+var d3 = require('d3');
+var callEach = require('lodash').callEach;
+var bindKey = require('lodash').bindKey;
+var rest = require('lodash').rest;
+
+function Binder($scope) {
+  this.disposal = [];
+  if ($scope) {
+    $scope.$on('$destroy', bindKey(this, 'destroy'));
+  }
+}
+
+Binder.prototype._bind = function (on, off, emitter, args) {
+  on.apply(emitter, args);
+  this.disposal.push(function () {
+    off.apply(emitter, args);
+  });
+};
+
+Binder.prototype.on = function (emitter/*, ...args */) {
```

Comment:
Can you ditch the commented out code?  We might as well not introduce commented out code in a brand new file.